syntax = "proto2";
option optimize_for = LITE_RUNTIME;
package YOUMEServiceProtocol;
import "youme_rc_comm.proto";


//关系链获取好友列表
message GetRcFriendListReq
{
	required int32 version = 1;
	required int32 start_idx = 2;		//分页起始小标 同mysql中limit
	required int32 len = 3;				//分页长度
}

//关系链获取好友列表回复
message GetRcFriendListRsp
{
	required int32 version = 1;
	required int32 ret = 2;
	required int32 list_cnt = 3;
	repeated YOUMEServiceProtocol.RcFriendBriefInfo friend_list = 4;		//好友列表
}

//关系链获取黑名单列表
message GetRcBlackListReq
{
	required int32 version = 1;
	required int32 start_idx = 2;		//分页起始小标 同mysql中limit
	required int32 len = 3;				//分页长度
}

//关系链获取黑名单列表回复
message GetRcBlackListRsp
{
	required int32 version = 1;
	required int32 ret = 2;
	required int32 list_cnt = 3;
	repeated YOUMEServiceProtocol.RcFriendBriefInfo black_list = 4;		//黑名单列表
}

//好友请求信息
message RcFriendReqInfo
{	
	required int32 app_id = 1;		
	required uint64 req_id = 2;	//不需要全局唯一，对于同一种请求,在两者之间唯一就好
	required RC_FRIEND_REQ_TYPE req_type = 3;
	required bytes myself = 4;	//请求在逻辑上的主语，比如我添加谁为好友，我被谁添加为好友
	optional bytes myself_nickname = 5;
	required bytes other_side = 6;
	optional bytes other_side_nickname = 7;
	required RC_FRIEND_REQ_STATE cur_state = 8;	//请求当前状态，对于请求的双方可能不一样
	optional bytes ext_info = 9;		//其他信息，比如验证字符串等
	optional int32 create_time = 10;	//创建时间
	optional int32 update_time = 11;	//最近更新时间
	optional int32 is_browsed_by_app = 12;	//是否已通知或拉取过
}

//获取好友请求列表
message GetRcFriendReqListReq
{
	required int32 version = 1;
	required int32 start_idx = 2;		//分页起始小标 同mysql中limit
	required int32 len = 3;				//分页长度
}

//获取好友请求列表回复
message GetRcFriendReqListRsp
{
	required int32 version = 1;
	required int32 ret = 2;
	required int32 list_cnt = 3;
	repeated RcFriendReqInfo friend_req_list = 4;		//好友列表
}


//发往自己的好友请求索引信息
message RcFriendReqIdx
{
 required int32 app_id = 1;  
 required uint64 req_id = 2; //不需要全局唯一，对于同一种请求,在两者之间唯一就好
 required RC_FRIEND_REQ_TYPE req_type = 3;
 required bytes other_side = 4;
}

//关系链APP浏览好友请求列表
message AppBrowseRcFriendReqListReq
{
	required int32 version = 1;
	repeated RcFriendReqIdx friend_req_list = 2;
}

//关系链APP浏览好友请求列表回复
message AppBrowseRcFriendReqListRsp
{
	required int32 version = 1;
	required int32 ret = 2;
}